### ISSUE TITLE
Is this connector supports avro format?

### ISSUE THREAD

**Issue by saurabh1920** [jcustenborder/kafka-connect-transform-common#62]

**Comment by jcustenborder** [comment]
Yes this does support Avro. Are you running into a problem with one of the transforms?

**Comment by saurabh1920** [comment]
Yeah, I am using this with mongo change stream. I want to rename key name which consist '$' in it.
I used this:
```
"transforms" : "patternRename",
"transforms.patternRename.type" : "com.github.jcustenborder.kafka.connect.transform.common.PatternRename$Value",
"transforms.patternRename.field.pattern" : "numberLong",
"transforms.patternRename.field.replacement" : "getout"
```

I am getting error 
`Jul 23 10:52:36 ip-172-32-1-97 connect-distributed.sh[7080]: Caused by: java.lang.UnsupportedOperationException: STRING is not a supported type.
`

Can you suggest where I am doing wrong?

**Comment by jcustenborder** [comment]
The mongo change stream connector doesn't support Avro. Basically each record that is coming out of mongo is getting passed through Kafka Connect as a String. Their connector is actually writing Schema.Type.STRING, "<bson of the record>" and handing that to Kafka Connect. This transformation requires a Struct or Map when receiving the message, that's why you're seeing the UnsupportedOperationException. You could look at trying to parse this string using [kafka-connect-json-schema](https://www.confluent.io/hub/jcustenborder/kafka-connect-json-schema), then apply the rename.
